Make compatible with Boot-2.0

artembilan opened the pull request:
Related to https://github.com/spring-cloud-stream-app-starters/app-starters-release/issues/134

* Perform appropriate changes for compatibility with the latest SI and Boot
* Change Gemfire dependency to Geode for Open Source Policy compatibility
* Add `geode-spring-boot-starter` dependency for better experience
* Add some dependencies "kung-fu" to make Geode compatible with Spring Boot
* downgrade to Log4J-2.8.2 - the compatible with Geode version
* Exclude `log4j-to-slf4j` which is not compatible with what Geode  uses

artembilan committed:
Make compatible with Boot-2.0

Related to https://github.com/spring-cloud-stream-app-starters/app-starters-release/issues/134

* Perform appropriate changes for compatibility with the latest SI and Boot
* Change Gemfire dependency to Geode for Open Source Policy compatibility
* Add `geode-spring-boot-starter` dependency for better experience
* Add some dependencies "kung-fu" to make Geode compatible with Spring Boot
* downgrade to Log4J-2.8.2 - the compatible with Geode version
* Exclude `log4j-to-slf4j` which is not compatible with what Geode  uses

artembilan commented:
NOTE: This PR introduced `geode-spring-boot-starter` dependency which is in active development and should be released in a week or though.

Let me know, @sabbyanandan , if that doesn't work for us and we definitely have to release next week.
Therefore I'll consider to PR without such an auto-configuration goodies.

// @jxblum

BTW, John, pay attention how I just tricked that `log4j-to-slf4j` in the dependency management:
```
<dependency>
	<groupId>org.apache.logging.log4j</groupId>
	<artifactId>log4j-to-slf4j</artifactId>
	<version>${log4j2.version}</version>
	<scope>provided</scope>
</dependency>
```
And generated uber jar won't have it any more!

Also with @tzolov 's help we realized that we don't need `log4j-slf4j-impl` as well. The provided factory by the `logback-classic` is fully enough.

Thank you everyone for the help!

sabbyanandan commented:
If it is only a week or little over, let's please wait and release the App then. If it is acceptable, we could release 2.0 M1 of this App standalone (after Darwin.M1 - it can still go next week) and have this App (2.0 RC1) join the train in Darwin.RC1.

artembilan committed:
Replace `geode-spring-boot-starter` dependency with an internal
`ClientCacheAutoConfiguration` implementation and direct `spring-data-geode` usage

artembilan committed:
* We don't need to add our internal conditional configuration to the `spring.factories`

jxblum commented:
Hi @artembilan - sorry for the delay on the _Spring Boot Data Geode/GemFire_ bits; I have been meaning to give you an update.  I got a bit delayed due to working on a customer issue.  I should be able to (hopefully) have the M1 bits prepared and finished by end of April.

tzolov commented on spring-cloud-starter-stream-processor-aggregator/src/main/java/org/springframework/cloud/stream/app/aggregator/processor/ExcludeStoresAutoConfigurationEnvironmentPostProcessor.java:
```diff
@@ -53,6 +56,7 @@ public void postProcessEnvironment(ConfigurableEnvironment environment, SpringAp
 						MongoRepositoriesAutoConfiguration.class.getName() + ", " +
 						EmbeddedMongoAutoConfiguration.class.getName() + ", " +
 						RedisAutoConfiguration.class.getName() + ", " +
+//					TODO	ClientCacheAutoConfiguration.class.getName() + ", " +
```
I guess the local `ClientCacheAutoConfiguration` (e.g. the temporal copy from Jonh) is not a real a real auto-configuration (e.g not in the spring.factories) and this is the reason to exclude it here?

tzolov replied:
Also, you can add 2018 to the license doc

artembilan replied:
Well, technically yes, it is not a fully-blown auto-configuration since it isn't included into the `spring.factories`. Agreed. And that is the reason why it's not included here in excludes (cofused? :smile: ), but at the same time we will need it when `gemfire` profile is selected.

I would appreciate if you fix that `2018` during merge :wink:

tzolov replied:
Thanks for the clarification! I will take care of the date in the merge

tzolov commented:
Please check the comments above.  
The changes LGTM though i haven't run e2e tests.

tzolov commented:
Rebased and merged at d3086c0
